File: CapeStoneProject/CapeStoneProject/src/main/java/com/capestone/model/PolicyAddOn.java
package com.capestone.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

@Entity
@Getter
@Setter
public class PolicyAddOn {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String name;

    private String description;

    private BigDecimal additionalCost;

    private Boolean active = true;

}

File: python_project/models/policy_add_on.py
class PolicyAddOn:
    def __init__(self, addon_id, name, description, additional_cost, active=True):
        self.__id = addon_id
        self.__name = name
        self.__description = description
        self.__additional_cost = float(additional_cost)
        self.__active = active

    @property
    def id(self):
        return self.__id

    @id.setter
    def id(self, value):
        self.__id = value

    @property
    def name(self):
        return self.__name

    @name.setter
    def name(self, value):
        self.__name = value

    @property
    def description(self):
        return self.__description

    @description.setter
    def description(self, value):
        self.__description = value

    @property
    def additional_cost(self):
        return self.__additional_cost

    @additional_cost.setter
    def additional_cost(self, value):
        self.__additional_cost = float(value)

    @property
    def active(self):
        return self.__active

    @active.setter
    def active(self, value):
        self.__active = bool(value)
